### PR TITLE
Support JDK 23

### PR DESCRIPTION
**Pull request by msridhar**
As usual, we verify existing tests don't crash on JDK 23 bytecodes, but we haven't added or tested support for any new features.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
           - os: ubuntu-latest
             java: 21
           - os: ubuntu-latest
-            java: 22
```
Comment (msridhar) [inline .github/workflows/continuous-integration.yml:30]: We don't need to do long-term testing on non-LTS releases so just switch 22 to 23

File: core/src/main/java/com/ibm/wala/analysis/reflection/GetMethodContext.java [hunk 2]
```diff
@@ -37,10 +37,12 @@
  * GetMethodContextSelector} should be placed in be placed in front of {@link
  * JavaLangClassContextInterpreter} and {@link JavaLangClassContextSelector} .
  *
+ * <p>TODO Do the same for {@link Class#getField(String)} and {@link
```
Comment (msridhar) [inline core/src/main/java/com/ibm/wala/analysis/reflection/GetMethodContext.java:40]: fixes a javadoc warning

File: core/src/main/java/com/ibm/wala/dataflow/IFDS/TabulationSolver.java [hunk 3]
```diff
@@ -53,8 +53,6 @@
  *   <li>it stores summary edges at each callee instead of at each call site.
  * </ul>
  *
- * <p>
```
Comment (msridhar) [inline core/src/main/java/com/ibm/wala/dataflow/IFDS/TabulationSolver.java:56]: fixes a javadoc warning

File: core/src/main/java/com/ibm/wala/ipa/callgraph/impl/ExplicitCallGraph.java [hunk 4]
```diff
@@ -226,10 +226,6 @@ protected boolean addTarget(int pc, CGNode tNode) {
       }
     }
 
-    /**
```
Comment (msridhar) [inline core/src/main/java/com/ibm/wala/ipa/callgraph/impl/ExplicitCallGraph.java:229]: for some reason this causes a javadoc error on 23; i can't understand why.  it's a useless comment anyway so i just deleted it

**Review by liblit — APPROVED** [review]

**Comment by msridhar** [comment]
Once this passes review I will update the required CI jobs.

**Pull request merged**